mvcc/kvstore_txn.go

```diff
@@ -124,6 +127,9 @@ func (tr *storeTxnRead) rangeKeys(key, end []byte, curRev int64, ro RangeOptions
 	}
 
 	revpairs := tr.s.kvindex.Revisions(key, end, rev)
+	if tr.trace != nil {
+		tr.trace.Step("Range keys from in-memory index tree.")
+	}
```

Comment:
Trace would only be `nil` for tests? I'm in favor of making trace a required parameter and removing these nil checks. That way we're running the same code paths in our tests that we will run in production.

Reply:
I think the nil check is necessary here. Since `rangeKey` is also used by `storeTxnWrite.Range()`
https://github.com/etcd-io/etcd/blob/129138760cbb79bc183abf6652455d27d6748818/mvcc/kvstore_txn.go#L79-L85
The trace object of `storeTxnWrite` is nil
https://github.com/YoyinZyc/etcd/blob/d9cf3d9cf460081b021eeb8762e5642bdc6f0389/mvcc/kvstore_txn.go#L72
The range of storeTxnWrite is for txn range request. I think it is out of the scope of the first pass that only trace range request.

Reply:
An alternative to the nil checks would be to pass in a noop trace where we don't need it.  That way we don't need to worry about forgetting a nil check and getting a nil pointer panic.